File: JobAdderChallenge/ClientApp/src/app/interfaces/job.ts
import { Candidate, TopCandidate } from './candidate';

export interface Job {
  id: number,
  name: string,
  company: string,
  topCandidate: Candidate
}

export interface JobDetail {
  id: number,
  name: string,
  company: string,
  skills: string[],
  topCandidates: TopCandidate[]
}

File: JobAdderChallenge/Controllers/Api/Models/Job.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace JobAdderChallenge.Controllers.Api.Models
{
    public class Job
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Company { get; set; }
        public Candidate TopCandidate { get; set; }
    }
}